### ISSUE TITLE
Only pull images that are not locally available yet

### ISSUE THREAD

**Issue by manuel-koch** [wagoodman/dive#82]
I can't hook/replace "docker build" command with "dive build", the command that builds the docker image is out of my control.

When docker image is build and I try to analyze the created image, "dive" initially tries to pull
the selected image unconditionally from external docker registry, despite the recent build/tagged image already being available locally !?

A command line switch to disable "pull" for locally available images is needed.
I.e. only pull selected image if it's not available locally yet.

**Comment by wagoodman** [comment]
Currently dive will only pull if it can't find the image locally. To help debug further: 
1. what OS are you using? 
2. what commands did you run to reproduce this?
3. can you post the output of `docker images` as well?

**Comment by manuel-koch** [comment]
Using dive 0.3.0 on MacOSX High Sierra via brew.

Building a docker image from local directory and a Dockerfile and adding tags ( e.g. latest ) too.
`docker image inspect gcr.io/my-namespace/foo:latest` outputs
```
$ docker image inspect gcr.io/my-namespace/foo:latest
[
    {
        "Id": "sha256:eb4f7c56760b2a94f19b7a77a40568595af58e38a13c0b5beabc65dc3f535a6d",
        "RepoTags": [
            "gcr.io/my-namespace/foo:latest"
        ],
...
```

`dive gcr.io/my-namespace/foo:latest` outputs
```
Analyzing Image
latest: Pulling from my-namespace/foo
Digest: sha256:142145bc2aa249d99e9e92025e1cdbfaccfa73d0af2713cfc534d4f28e3bba94
Status: Downloaded newer image for gcr.io/my-namespace/foo:latest
  Fetching metadata...
  Fetching image... [>                                        ] 1 % (7275520/727480637)
  Fetching image... [=>                                       ] 3 % (21824512/727480637)
  Fetching image... [====================>                    ] 51 % (371015680/727480637)
  Fetching image... [======================>                  ] 56 % (407389184/727480637)
  Fetching image... [=======================>                 ] 59 % (429213696/727480637)
  Fetching image... [========================================>] 100 % (727480637/727480637)
  Discovering layers... Done!
    ├─ f72ac8392a3dae5 : [========================================>] 100 % (3/3)
    ├─ f1b8c89df994ceb : [========================================>] 100 % (319/319)
    ├─ c1a05033b85ad4e : [========================================>] 100 % (3/3)
    ├─ c05cfc7cfe22ea5 : [========================================>] 100 % (4/4)
    ├─ b501d078d80aa1d : [========================================>] 100 % (5/5)
    ├─ b4ee25a6736989e : [========================================>] 100 % (3946/3946)
    ├─ af2cdc82ca8b09a : [========================================>] 100 % (11540/11540)
    ├─ a921b5e63f733c7 : [========================================>] 100 % (3/3)
    ├─ a84ce3157d347f5 : [========================================>] 100 % (20/20)
    ├─ a7b4753b5f0e2e8 : [========================================>] 100 % (4559/4559)
    ├─ a04eb1dbfae0e1c : [========================================>] 100 % (14446/14446)
    ├─ 8f11be67b051b24 : [========================================>] 100 % (1759/1759)
    ├─ 5d68b369e5f7045 : [========================================>] 100 % (8/8)
    ├─ 4047c867de8704e : [========================================>] 100 % (636/636)
    ├─ 3efa53502e57a4e : [========================================>] 100 % (5070/5070)
    ├─ 3cc9c1f4cdcc7ed : [========================================>] 100 % (5/5)
    ├─ 122cb1e4cdedf00 : [========================================>] 100 % (7/7)
    ├─ 0be384e5023f524 : [========================================>] 100 % (1247/1247)
    ├─ 021b7c932d118de : [========================================>] 100 % (721/721)
  Fetching image config...
  Building tree...
  Analyzing layers...
```

dive is inspecting a different image !?

My local build/tagged image is 
sha256:eb4f7c56760b2a94f19b7a77a40568595af58e38a13c0b5beabc65dc3f535a6d
dive is inspecting image
sha256:142145bc2aa249d99e9e92025e1cdbfaccfa73d0af2713cfc534d4f28e3bba94

**Comment by manuel-koch** [comment]
Using `dive 0.3.0` installed via `brew` on MacOSX High Sierra.

Building a docker image from directory using Dockerfile, adding tags like "latest" to it.

`$ docker image inspect gcr.io/my-namespace/foo:latest`
outputs
```
[
    {
        "Id": "sha256:eb4f7c56760b2a94f19b7a77a40568595af58e38a13c0b5beabc65dc3f535a6d",
        "RepoTags": [
            "gcr.io/my-namespace/foo:latest"
        ],
        "RepoDigests": [
            "gcr.io/my-namespace/foo@sha256:142145bc2aa249d99e9e92025e1cdbfaccfa73d0af2713cfc534d4f28e3bba94"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2018-11-18T12:20:53.096986815Z",
        "Container": "cb422b29f6e9f0788e4c5ee962815b964da6a02cde0f6681f9707cafd043172e",
        "ContainerConfig": {
            "Hostname": "cb422b29f6e9",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                ....
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "LABEL APP_VERSION=e3e4c961 io.k8s.description=e3e4c961 version=e3e4c961 release=e3e4c961"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:fd35ee6aac8cd3eb902a6d216259c80d5b218fd08ef1e72813a366b0187ddf4e",
            "Volumes": null,
            "WorkingDir": "/opt/app-root/app",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                ....
            }
        },
        "DockerVersion": "18.06.1-ce",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                ....
            ],
            "Cmd": [
                "/opt/app-root/run.sh"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:fd35ee6aac8cd3eb902a6d216259c80d5b218fd08ef1e72813a366b0187ddf4e",
            "Volumes": null,
            "WorkingDir": "/opt/app-root/app",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                ....
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 727480637,
        "VirtualSize": 727480637,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/dd94589152397d1e677906a3908d7b2b10c09d917fd3cd5cd8d53621c1d14a85/diff:/var/lib/docker/overlay2/b9cb87093a5d282765b9d5fb0c1c610a9f1639bcba9a65fb0c46af076803b081/diff:/var/lib/docker/overlay2/5962ad6a44b643f32d6b7691903e4eb5b7419869cdeedd999f366b3b8347fb97/diff:/var/lib/docker/overlay2/2cef212cbf084e5e0c8e3de873f5faf07690a966bb2c7a8c3bb611c6ba18daa9/diff:/var/lib/docker/overlay2/ad0222eb58d6436c8cfb84702cd70e47dbdca5848b444a1ea4886172e52aa4ec/diff:/var/lib/docker/overlay2/ffbe32b49cf8e199001c5167eb00a21f62d6ea32a8197a98d9325a289bc54b8d/diff:/var/lib/docker/overlay2/f1bcf647567688f12291dc40df97d00dfb56b6f528c3d648cd9df740e3b3fbcc/diff:/var/lib/docker/overlay2/9b4c58d01c29776842f403ffda347ee80e8fabcd052f11e243012a89945b4cba/diff:/var/lib/docker/overlay2/3d2849fdcb5c86fb2cb3c58250875c9425989a8984f9acc5cac8d1e53043f71b/diff:/var/lib/docker/overlay2/754708e49d82fd2e2701055ac032e30035f0df032282c9ae08415600715b0545/diff:/var/lib/docker/overlay2/7224eff69e4bf02c812d88ad5ea406eb4180d99fe4147b884fbf63994b616a09/diff:/var/lib/docker/overlay2/f67740015c6f3e0374b068408a6d6823af4d63c09acbb22d910310ef3b317978/diff:/var/lib/docker/overlay2/b8d9deaa17acb3886f5a3366e46ac362f1d8974882de9d1074cdd7e29b429cf9/diff:/var/lib/docker/overlay2/654037a2ad969f20ed331126796e5dd8e6f638b91531923a3b06e15dc670d7bf/diff:/var/lib/docker/overlay2/c8412e66139ff280722c74e05809a40a6fb54e6ef35619f5a5706f26b1f67148/diff:/var/lib/docker/overlay2/74c6e7e5cefee2dc4793a1658185273921e3a8e325f397826a3d2495a2a423d6/diff:/var/lib/docker/overlay2/a4d2fbdc9d1e4a7c4d24c215e3a79ff3bb738e602dcfcb2a551d074e0e377f2a/diff:/var/lib/docker/overlay2/f801d14a6b1d778bf98de76be9a26b4493c874e0e64d10fac21b0003a0c2f2a4/diff",
                "MergedDir": "/var/lib/docker/overlay2/76d986d70f21dd01b38460b776c3cdb40ce0bc949f6ba200d6601dbd8e7a8719/merged",
                "UpperDir": "/var/lib/docker/overlay2/76d986d70f21dd01b38460b776c3cdb40ce0bc949f6ba200d6601dbd8e7a8719/diff",
                "WorkDir": "/var/lib/docker/overlay2/76d986d70f21dd01b38460b776c3cdb40ce0bc949f6ba200d6601dbd8e7a8719/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:237472299760d6726d376385edd9e79c310fe91d794bc9870d038417d448c2d5",
                "sha256:dd6cb05ac6a5d8ceb28bcf3005e7d6a46137d7d6b84e60f19526eefe9e5e3d56",
                "sha256:9a20de67e15da15c27dca4796ad61d25857a20ce605ec94a18f066544b09200e",
                "sha256:684892fc15c076e901f6be622b55713707263620cbb2a55a94eef1d92566eb40",
                "sha256:19a7af556a81d32c9d701bfaf6210cc9f7777c76bcb1efe86dbd141fe1171d64",
                "sha256:02b51a0cf7b1f4e71bec6decdb7c0e52e8c9bcc82acdfef891b09a89e0cef2f5",
                "sha256:299cb3066dfeeb4ab2dbc539f5d9c3b8c46011efba65f6550a8a45b31fd6d615",
                "sha256:33ee6b4232335914c41aa13f289deb9672969da7b7531eab6b3682963cd83a56",
                "sha256:b2c0b0657629c2ab3cffb39b15431b243af42999bb7509d5cee26ba9c7ad384d",
                "sha256:a389353489dadc435774eb71d7267abbf6a7e089ef450131574d5513b9ab3212",
                "sha256:badd834ed37b7706e6aa885d5da0a1c2fd2c5365d7057e0e71329eeacea65842",
                "sha256:a3487f7abaf3221a1b75aaed475f85701b70d41b7bd759195faf207b76549537",
                "sha256:7640b367d27fc624c8eb38350429b50715968a606dcea22347910f26641cfc43",
                "sha256:341c65ca36ebb5d8dd05aa7474a2333b26c745243551a099a00d701fa148398d",
                "sha256:347ea96acc71ecbc5f6124e8db71e0a2e94eb15a9c8b8ea92d7485fcb9a5ab92",
                "sha256:786d0e7fc8f3e695f109b334bad62cd51af05d8cfb18ae6cb7e46a722463f39e",
                "sha256:cf5ee7709ccfc79b1714f5b56743cd3c68f8d3097034d8d5e95ebea820eda0a3",
                "sha256:4462f18b5dec29b7ea6d265c4ef1abb1ccd613b6d86c4070014415897a09b06a",
                "sha256:be95f25019aaf69e61de7ee4749a5e5649f2afe18a21b7d73c8c500134394c60"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

`$ dive gcr.io/my-namespace/foo:latest`
outputs
```
Analyzing Image
latest: Pulling from my-namespace/foo
Digest: sha256:142145bc2aa249d99e9e92025e1cdbfaccfa73d0af2713cfc534d4f28e3bba94
Status: Image is up to date for gcr.io/my-namespace/foo:latest
  Fetching metadata...
  Fetching image... [========================================>] 100 % (727480637/727480637)
  Discovering layers... Done!
    ├─ f72ac8392a3dae5 : [========================================>] 100 % (3/3)
    ├─ f1b8c89df994ceb : [========================================>] 100 % (319/319)
    ├─ c1a05033b85ad4e : [========================================>] 100 % (3/3)
    ├─ c05cfc7cfe22ea5 : [========================================>] 100 % (4/4)
    ├─ b501d078d80aa1d : [========================================>] 100 % (5/5)
    ├─ b4ee25a6736989e : [========================================>] 100 % (3946/3946)
    ├─ af2cdc82ca8b09a : [========================================>] 100 % (11540/11540)
    ├─ a921b5e63f733c7 : [========================================>] 100 % (3/3)
    ├─ a84ce3157d347f5 : [========================================>] 100 % (20/20)
    ├─ a7b4753b5f0e2e8 : [========================================>] 100 % (4559/4559)
    ├─ a04eb1dbfae0e1c : [========================================>] 100 % (14446/14446)
    ├─ 8f11be67b051b24 : [========================================>] 100 % (1759/1759)
    ├─ 5d68b369e5f7045 : [========================================>] 100 % (8/8)
    ├─ 4047c867de8704e : [========================================>] 100 % (636/636)
    ├─ 3efa53502e57a4e : [========================================>] 100 % (5070/5070)
    ├─ 3cc9c1f4cdcc7ed : [========================================>] 100 % (5/5)
    ├─ 122cb1e4cdedf00 : [========================================>] 100 % (7/7)
    ├─ 0be384e5023f524 : [========================================>] 100 % (1247/1247)
    ├─ 021b7c932d118de : [========================================>] 100 % (721/721)
  Fetching image config...
  Building tree...
  Analyzing layers...
```

Looking at the file contents of the image with `dive` shows files of an "old" build that was pushed to the remote registry using the "latest" tag !?

**Comment by ccinelli** [comment]
I have the same problem. I think it may be when you use a private repository:
```
10:32 $ dive user/repo:latest
Analyzing Image
Error response from daemon: pull access denied for user/repo, repository does not exist or may require 'docker login'
  Fetching metadata...
  Fetching image... [========================================>] 100 % (1570175458/1570175458)
  Discovering layers... Done!
    ├─ dd605190e5708b5 : [========================================>] 100 % (16/16)
    ├─ 939c459c58cb35c : [========================================>] 100 % (4/4)
...
```

**Comment by ccinelli** [comment]
Making sure that the message specify when it is reading from a **local** copy.

**Comment by ChrisCinelli** [comment]
This should be fixed but #104

**Comment by wagoodman** [comment]
#104 has been merged, this should make it in the next release
